### DISCUSSION TITLE
useMutationState restart data

### DISCUSSION THREAD

**Discussion by jgalianoz** [TanStack/query#6858] [General]
Is there a way to reset the data of useMutationState since every mutation I make adds one more object to my response?

<img width="350" alt="image" src="https://github.com/TanStack/query/assets/12010942/a312f711-a7aa-4e31-8cb1-0aa1241ca7e3">

**Comment by TkDodo** [comment]
you can provide filters

**Reply by KarthikRaju391** [reply]
I have a list of items and on adding a new item, I want to focus on the element that was added. 
So, I thought I could use the mutationState to know when a new creation happens, the length of the mutationState will be 1 and if it's 1, I am using the ref to focus on that element. 

But, once, that `focus` is done, I want to reset the mutationState so that my check for the length of the mutationState to be 1 works only when new items are added.
Is there a way I could do that? Or is this approach wrong altogether?

**Reply by TkDodo** [reply]
that's what `useMutation` usually does per default.

**Reply by KarthikRaju391** [reply]
For every mutation, the length of the mutationState is increasing similar to the example shown by @jgalianoz. I just want to know if it's possible to reset the mutation state manually so that, when the new item is added and the focus is done, I can clear the mutation state.

**Reply by TkDodo** [reply]
mutations are kept in the mutationCache for `gcTime` milliseconds, which defaults to 5 minutes. If you set it to 0, mutations will be removed as soon as they become un-observed, which happens when you fire another mutation from the same `useMutation` call

**Reply by KarthikRaju391** [reply]
Got it, thank you!

**Comment by ojj1123** [comment]
@TkDodo 

I agree with `useMutationState` return `Array<TResult>` since there can be multiple things that get filtered. But I expected that every mutation reset the data of `useMutationState` being same `mutationKey`, it doesn't.
I was wondering if you could explain why the mutation state that have the same `mutationKey` are cached **separately**:

<img width="383" alt="스크린샷 2024-04-30 15 35 16" src="https://github.com/TanStack/query/assets/33178048/89bfd197-8450-41fb-a799-9e1fd3b99ddd">

**Reply by TkDodo** [reply]
because every invocation of `mutate` creates a new mutation that is not tied to the other one. Mutations work differently from queries - they don't deduplicate with the same key.
